docs/sources/configure-client/grafana-agent/go_pull.md

```diff
@@ -1,258 +1,224 @@
 ---
 title: "Go (pull mode)"
 menuTitle: "Go (pull mode)"
-description:  "Instrumenting Golang applications for continuous profiling"
+description: "Set up Go profiling in pull mode"
 weight: 10
 ---
 
 # Go (pull mode)
 
-The Grafana Agent is responsible for pulling profiles from applications and delivering them
-to the Pyroscope database.
+In pull mode, the Grafana Agent periodically retrieves profiles from Golang applications, specifically targeting the
+`/debug/pprof/*` endpoints.
 
-### Prerequisites
+## Set up Go profiling in pull mode
```

Comment:
Really good changes! Love the steps you've added and the matching headings. 

We probably don't need a heading here since the content below is really part of the introduction. However, are there any prerequisites that people need to have before continguing into the procedures? If so, consider adding a "Before you begin" section that lists those requirements. As an example, refer to https://grafana.com/docs/tempo/latest/setup/tanka/
```suggestion
```

Reply:
The only prerequisite is to have grafana agent, and I put it not as prerequisite so not needed